src/libnml/cms/cms_cfg.cc

```diff
@@ -305,7 +305,7 @@ int cms_copy(CMS ** dest, CMS * src, int set_to_server, int set_to_master)
 extern char *get_buffer_line(const char *bufname, const char *filename)
 {
     int line_len, line_number;
-    char linebuf[CMS_CONFIG_LINELEN];	/* Temporary buffer for line from
+    static char linebuf[CMS_CONFIG_LINELEN];	/* Temporary buffer for line from
```

Comment:
Why do you make this variable static?  Why should its content survive from call to call?

Reply:
Happily, this function is deleted in the current version!

Originally, a pointer within `linebuf` was returned from the function, which is undefined behavior since the storage scope of `linebuf` ends when the function returns:
```c
    char linebuf[LINELEN];	/* Temporary buffer for line from
					   file. */
    char *line = linebuf;
    // …
   return line;
```
By making it static, the storage remains valid for the lifetime of the program, though it is over-written by a subsequent call to `get_buffer_line`, so some care is still required in its use.

Removal: 35659250289e74ee15344885cba3ff91cb9b2dde (thanks @dwrobel !)

Reply:
Dropped on rebase.